packages/protocolbeat/src/panel-nodes/controls/node-values/ValuesDialog.tsx

```diff
@@ -0,0 +1,123 @@
+import * as Dialog from '@radix-ui/react-dialog'
+import { type SVGProps, useCallback, useMemo, useState } from 'react'
+import type { Node } from '../../store/State'
+import { useStore } from '../../store/store'
+import { ControlButton } from '../ControlButton'
+import { FieldsList } from './FieldsList'
+import { groupJsonFields } from './groupJsonFields'
+
+export const ValuesDialog = {
+  Root: ValuesDialogRoot,
+  Trigger: ValuesDialogTrigger,
+  Body: ValuesDialogBody,
+}
+
+function ValuesDialogRoot({ children }: { children: React.ReactNode }) {
+  return <Dialog.Root>{children}</Dialog.Root>
+}
+
+function ValuesDialogTrigger({ disabled }: { disabled: boolean }) {
+  return (
+    <Dialog.Trigger asChild disabled={disabled}>
+      <ControlButton disabled={disabled}>Values</ControlButton>
+    </Dialog.Trigger>
+  )
+}
+
+function ValuesDialogBody({ node }: { node: Node }) {
+  const setNodes = useStore((state) => state.setNodes)
+  const nodes = useStore((state) => state.nodes)
+
+  const [hiddenFields, setHiddenFields] = useState(node.hiddenFields)
+  const groupedFields = useMemo(
+    () => groupJsonFields(node.fields),
+    [node.fields],
+  )
+
+  const modifyNode = useCallback(() => {
+    const newNode = {
+      ...node,
+      hiddenFields,
+    }
+
+    setNodes(nodes.map((n) => (n.id === node.id ? newNode : n)))
+  }, [node, hiddenFields, setNodes])
+
+  return (
```

Comment:
Add three buttons:

- All (selects all)
- None (unselects all)
- Invert (inverts the selection)